File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityLoading.java
package com.tamsiree.rxdemo.activity;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;
import com.tamsiree.rxdemo.R;
import com.tamsiree.rxdemo.fragment.FragmentLoadingDemo;
import com.tamsiree.rxdemo.fragment.FragmentLoadingWay;
import com.tamsiree.rxtool.RxDeviceTool;
import com.tamsiree.rxui.activity.ActivityBase;
import com.tamsiree.rxui.view.RxTitle;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author tamsiree
 */
public class ActivityLoading extends ActivityBase {

    @BindView(R.id.rx_title)
    RxTitle mRxTitle;
    @BindView(R.id.tabs)
    TabLayout mTabs;
    @BindView(R.id.viewpager)
    ViewPager mViewpager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_loading);
        ButterKnife.bind(this);
        RxDeviceTool.setPortrait(this);
        mRxTitle.setLeftFinish(mContext);
        mViewpager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {

            String[] titles = new String[]{
                    "加载的方式", "加载的例子"
            };

            @Override
            public Fragment getItem(int position) {
                if (position == 0) {
                    return FragmentLoadingWay.newInstance();
                } else {
                    return FragmentLoadingDemo.newInstance();
                }
            }

            @Override
            public int getCount() {
                return 2;
            }

            @Override
            public CharSequence getPageTitle(int position) {
                return titles[position];
            }
        });
        mTabs.setupWithViewPager(mViewpager);
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityLoading.kt
package com.tamsiree.rxdemo.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import com.tamsiree.rxdemo.R
import com.tamsiree.rxdemo.fragment.FragmentLoadingDemo
import com.tamsiree.rxdemo.fragment.FragmentLoadingWay
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxui.activity.ActivityBase
import kotlinx.android.synthetic.main.activity_loading.*

/**
 * @author tamsiree
 */
class ActivityLoading : ActivityBase() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_loading)
        setPortrait(this)
    }

    override fun initView() {
        rx_title.setLeftFinish(mContext)
        viewpager.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
            var titles = arrayOf(
                    "加载的方式", "加载的例子"
            )

            override fun getItem(position: Int): Fragment {
                return if (position == 0) {
                    FragmentLoadingWay.newInstance()
                } else {
                    FragmentLoadingDemo.newInstance()
                }
            }

            override fun getCount(): Int {
                return 2
            }

            override fun getPageTitle(position: Int): CharSequence? {
                return titles[position]
            }
        }
        tabs.setupWithViewPager(viewpager)
    }

    override fun initData() {

    }
}